pydax/_dataset.py

```diff
@@ -174,6 +175,7 @@ def load(self,
                 raise FileNotFoundError(f'Failed to load subdataset "{subdataset}" because some files are not found. '
                                         f'Did you forget to call {self.__class__.__name__}.download()?\nCaused by:\n'
                                         f'{e}')
```

Comment:
We should think about whether a half-loaded data object is allowed, i.e., whether we should insert `self._data = None` here.

Reply:
Hm good point, is there a specific disadvantage you think to returning a half-loaded data object,  or is the main concern that the user may just not expect that behavior if the `FileNotFoundError` triggers. If just the latter, then we should do whatever you think the average user would typically expect in this situation.

Reply:
I think it would be a great source of confusion if a half-loaded object is stored. I would say let's just set it to `None`.